### ISSUE TITLE
Some (minor) things about Duchess

### ISSUE THREAD

**Issue by iblowmymind** [devhawala/dwarf#4]
Hi there!
First of all, I've been really interested in Mesa and other related stuff recently. When I found this emulator, I was really amazed by what you managed to do, and suprised that this wasn't really well-known on the internet. The amount of progress in this emulator is really good! But, I have some minor questions about the Duchess emulator:

I installed GVWIN 2.1 on a Windows 3.1 virtual machine last year, used some registration keys I found (I think they came with the archive) to activate mostly everything (and I also did the PACKINFO modification or whatever it was called to include the rest) so I have a pretty complete image now. I used Dodo to hook it up to a virtual XNS network to demonstrate the thing, and it works really good! But, there's just a really really minor thing: The help icon on the top right doesn't show up. Now I know, this's a really minor concern, but I noticed that in the screenshot you provided it displays, so did I do something wrong while installing? Also a suggestion: If you could add a mode (maybe a configuration flag) to make the thing fullscreen, I think it'd be pretty cool, for example since there's a button on the top left that serves no purpose on the emulator but did serve a purpose in the original GVWIN where it'd display the windows menu. So it seems to me that fullscreen would be a good thing. I don't know. Like I said (for the 1000th time) these are really minor things.

You can ignore this if you'd like, but I just wanted to say that I really love both Draco and Duchess!

And also one final question: I was checking the other issues when I saw something related to a "GV developer mode". What is this, and does it work?

**Comment by devhawala** [comment]
Hello,

the problem with the help icon is strange, as i suppose the same disk did show the help icon when running with the original emulator under Win3.1. There were several bugs in my emulator resp. strange usages of the BitBlt instruction in GV2.1 that manifested in odd-looking help icons, but i thought all were fixed now. In all these cases, "something" was visible instead of the icon. Does the MessageBar at the upper border leave the place for the help icon? In other words: is there simply the background pattern where the help icon should be or does the MessageBar extend to the right? If the place for the help icon is not reserved, then the Help application may be not started, possibly set to autorun = false in the "Software Loader", however very inprobable...

For the fullscreen feature, i thought of it at the beginning and then postponed it, because of technical challenges at the detail level (does Java8 support putting a window above (instead of) the icon-/taskbar on all OSes, where to put / how to handle functionality currently in the menu bar (mount/dismount floppies, start/stop), how to get the effective screen size). And so display sizing is still like in the first version... Maybe some day (color was also postponed initially and is now there). Currently i work more on Dodo, currently for a mail service.

The "developer mode" allows to access lower level file system functions in VP/GV like changing the file type in the properties sheet, extended software loader (access invisible applications and programs, change more application properties), use additional (but unavailable) tools for accessing internal folders of the system. The problem is to enable it, as one cannot make the necessary change to the "Systemprofile" file without already being in developer mode! At the TU Berlin in the 80-ies, we edited the systemprofile from the XDE installed in parallel (which is not there in GVWIN 2.1 installations) or "installed" a patched systemprofile from a file server using the network installer (which is not present  for the GVWin-architecture either). But without the development tools like compiler, binder, programming interfaces etc., the development mode for VP/GV is worthless. Same for the "XDE on GV" application which i suppose would provide an empty XDE environment in VP/GV.

Greetings, Hans

**Comment by iblowmymind** [comment]
Ah, I see. Oddly, place for the help icon is reserved on the bar, and checking the loader (as far as I can remember) confirmed that it was running. One possible solution for the issue would be reinstalling it on a virtual machine and grabbing the disk from that, as I have the installer (I got it from archive.org afaik) but I don’t want to lose this image since I don’t exactly remember how to install the packages which don’t have a product key for (I have the keys for the basic software, fonts etc. but not stuff like arabic, hebrew, XPI and more advanced fonts), I remember it involves swapping a product info file but I don’t remember when. 

About the fullscreen, I’m not the best at Java but I think a fullscreen feature exists. And also, I’ve been missing a mail service in Dodo, so good luck on that!

And oh, I see. When I was experimenting with Darkstar in Bitsavers i saw a “GV Expert” package, upon launching it’d give me an XDE environment. So I suppose I already experienced that! And yeah, the developer tools are probably either long lost, lying in an old PARC storage room with other software, or probably in someone else’s attic, and the person has no idea, since they’re not available, like, anywhere, for both the 8010 nor 6085.

Thank you!

**Comment by devhawala** [comment]
Hello,

i remember to have downloaded the setup for GlobalView 2.1 from the archive.org page smoe years ago before starting the Dwarf development (however not sure), but today, i did not find a setup anymore (maybe the wrong search criteria), instead i found 2 ready to run DosBox with Win3.1 environments, with the same upload date but different times. Taking the GVWIN001.DSK and .GRM files in any of the downloaded ZIP files (one is called globalview.zip, the other gvwin.zip), one gets basically the (almost) same GlobalView 2.1 environment.

Adding the .jar, .sh and .properties file from the last Dwarf distribution as well as the keyboard-maps directory, i could start GlobalView and got the Help icon in the top right corner after logon (user _user:_ with password _pass_. Setting "Autorun at startup" to no and restarting the system removes the help icon as expected, here the screen shots from running the disk from globalview.zip in color mode display:
![globalview-help-running](https://user-images.githubusercontent.com/34173994/99428333-c9f8cb00-2906-11eb-8202-62d91b40a7f7.png)
vs. 
![globalview-help-stopped](https://user-images.githubusercontent.com/34173994/99428354-d2510600-2906-11eb-95ed-1e26f449e637.png)

Same when running the disk in black&white mot, i even tried a 640x480 color resolution (of course with the Help application started), the icon still showed up.
So i don't understand what is going wrong with your disk.

For a re-installation: if we are talking about the same setup-ZIP, there was an additional directory with a readme and a replacement package description file, i remember that the readme explained how to use the replacement file to install all packages. However i don't remember where i left that setup or the extracted files, i did not find it today :-(

Thanks for the hint about the "GV Expert" package, i'll give it a try some day and see what happens...

Greetings, Hans

**Comment by iblowmymind** [comment]
Hello,
Just a quick note: I think I might've gotten the installer from Winworldpc, since I remember it was there. And that's pretty interesting that the help icon's messing up. And also, a question: Is it technically possible to modify the systemprofile if you manually dump the disk contents, edit it, and put it back together somehow? Or maybe modifying the actual code that generates the profile file (probably the original GV 2.1 installer? Or a template file? If it's the latter, then that'd be much easier to do.)

**Comment by devhawala** [comment]
Hello,

your question about the developer mode made me think about it again...
As there is a working file service now and there are some development tools for Viewpoint at Bitsavers, there is a much simpler method to enable the developer mode, which i verified by experimenting a bit.

Here are the steps (hoping i will not forget something, this was tested in the globalview system found at archive.org, running with Duchess):
- backup your globalview system (DSK and all zdelta files)
- download the archive "6085_XDE_6.0_VP_Devel.zip" from Bitsavers
- unpack it in the "floppies" directory of your Globalview system
- make sure that Dodo including the Filing service is started
- start the Duchess system with globalview (the following must happen in the same boot session!)
- logon to globalview with a CHS user (e.g. motu:dev:hawala)
- copy the "Floppy Drive" icon from the directory to the desktop
- copy the icon of your favorite file drawer to the desktop
- copy the "Loader" icon from the directory to the desktop
- insert the file "tools.dmk" from the unpacked ZIP archve into the floppy drive
- open the floppy icon and copy the file "SystemFolder.bcd" to the desktop
- rename this file to "SystemFolder.autorun" using the property sheet
- open the Loader icon
- copy the SystemFolder.autorun file into the Loader (this can take a while!) but do not close the Loader!
- the copied file in the Loader should be selected even if not visible, simply hit "Run" in the title area of  the Loader window (possibly scroll to find it, select it and then hit "Run", but do not close the Loader, as this file will not be displayed when re-opening the Loader)
- the system menu in the top message bar should now have an item "System folder", choose this item
- the new window shows the system folder of the VP/GV system, scroll down to the end where the file "WorkstationProfile" should be visible
- copy this WorkstationProfile icon to your file drawer
- in the OS file system, go to the filing volume directory of your Dodo setup, open the youngest "files-000..." directory
- the youngest file there should be the content ot the "WorkstationProfile" that was copied to the file drawer
- edit this file and and do the following changes to activate the developer mode:
  -- in section [Application Loader] in the line "Developer:" change FALSE to TRUE
  -- in section [System] in the line "Developer:" change FALSE to TRUE
  -- save and close the file
- back in GV again in the still opened "System folder" window rename the file "WorkstationProfile" to "WorkstationProfile.original" (for example)
- copy the file "WorkstationProfile" from your file drawer to the "System folder" window
- close all windows
- shutdown Globalview normally
- restart globalview and logon again
- the developer mode should now be active, which can be noticed for example (as in my system now):
  -- in the Loader: you can see "invisible" applications (there is a new column for this)
  -- in the Loader: the .bcd and .autorun files are visible
  -- the property sheet for programs has more options
  -- the property sheet for normal files has the possibility to change the file type (mind the warning displayed)
  -- text files (and the WorkstationProfile) can be edited ("Edit" button in the title area)

Hope that helps you to enable the developer mode.

For the Help icon: can you post a screenshot of the upper right corner of your running globalview system to give me a hint what may go wrong?

Thanks and good luck
Hans

**Comment by iblowmymind** [comment]
Oh, thank you a lot for the guide! And sure, I'll post it here:
![image](https://user-images.githubusercontent.com/28228415/99669600-5a0e5000-2a80-11eb-8c2d-1f2df153dd3e.png)
Interestingly, last time I was sure it was running. So I checked the loader, and although the autostart is on, it doesn't start and asks for File Access. When I went to check that, oddly there was no File Access. Did I accidentslly delete it somehow?

**Comment by iblowmymind** [comment]
Oh and also, the developer mode works like a charm! And something I noticed is that when you do System Menu > Open with Simple Editor (which I'd assume opens the Tajo version of Editor?) leaves you in opcode 2822 with a debugger message like this: 
![image](https://user-images.githubusercontent.com/28228415/99671473-22ed6e00-2a83-11eb-8ce2-101957ef389e.png)
Interesting behaviour. Is it because it requires XDE in a different partition and opening with simple editor swaps worlds? (which is not implemented on Dwarf yet so that'd make sense that it gives an error)

**Comment by devhawala** [comment]
Hello,

there is no application "File access", but looking a bit further down in the Loader list, there is a software called "PC File Access" (version "GVWin 2.1").
After disabling "Auto Run at system startup" on "PC File Access" and re-booting GlobalView: bingo, the help icon is missing. Trying to run the Help application manually tells that "File Access" is not running, so simply a typo?
Please look if your "PC File Access" has autostart disabled. This program is probably for accessing files on the DOS/Windows host where the original GVWin 2.1 emulator would run, so it is useless when running the disk with Duchess, but apparently necessary for the system even if not used. However i have no idea how this program got disabled...

For the "Open with Simple Editor" problem: no, it probably does not require a parallel installation of XDE on the same disk. I remember that this function must be used with care, as it has a limit to 64K text length and may only be used for text files, but the program does not check this; using it with binary files or directories can end in disaster (falling back into local world-swap or network debugging, in your case it was the default debugger substitute of GV/VP). Maybe the GVWin version has more bugs than the old ViewPoint version 1.1 we had at TU Berlin...
(but you are right: developer mode _is_ intended to be used with an XDE installed in parallel)

Greetings
Hans

**Comment by iblowmymind** [comment]
Oh yeah, I remember now that I disabled PC File Access since it didn’t have any use anyways, then couldn’t find it. Thanks for the tip!

And hmm, I see. I tried it with the Workstationprofile to test it (which should be under 64K if I’m not wrong, since I checked it with the Dodo blocksize and it was somewhere between 10 and 20K) but it still crashed.

**Comment by iblowmymind** [comment]
Hey! So, a quick update on the Developer thing:
I enabled developer mode on that preinstalled VP2.0.5 disk, and inspected some files there too. Then last time I realised that I tried to use the simple editor with a folder by accident. I thought I selected the Workstationprofile. So then out of curiosity I started inspecting the XDE6 VP Developer floppies, and found this: 
[BWSOperationsGuide.doc.txt](https://github.com/devhawala/dwarf/files/5577740/BWSOperationsGuide.doc.txt)
The BWS Operations Guide! This's really interesting. Also in the tools disk I found the appropriate tools to create a master file/folder, convert application to folder etc. Also a ton of source code on the other floppies! (Sadly no compiler though.) And also stuff that I'm wondering: What's the GV 3.2.38 on the Bitsavers archive? Wasn't GV only for Windows/Solaris and OS/2? And why's it 3.x?
Also, is there any way to access the System partition (that contains the BWS, Germ, bootfiles etc.) from VP?

**Comment by devhawala** [comment]
Hello,

in fact the "6085_XDE_6.0_VP_Devel.zip" package is an add-on package for an XDE 6.0 development system to allow developing VP 2.0 applications (most files are progamming interfaces and symbols for debugging), sadly the 2 required packages did not make it to Bitsavers: the "XDE 6.0" and "Mesa Development Environment 6.0" floppy sets, where the latter has the Mesa compiler, binder and other development tools necessary to use the items in the VP 2.0 development package... (the XDE 5.0 package is useless, as there was an architectural change from Pilot 12.x (XDE5.0/VP1.1) to Pilot 14.0 (XDE6.0/VP2.0) which forces XDE6.0 when debugging VP2.0)
But the tools can be used inside VP2.0, as well as most VP1.x programs found in other XDE floppy sets at bitsavers (one or both of the 8010 XDE 5.0 sets have more programs than the pure XDE 5.0 installation kits in the 6085 floppy set)

The Globalview 3.2.8 package is an update package, requiring an installed version 3.2 of Globalview for 6085 (which is not available AFAIK), according to the release notes and a lot of internal files are missing on the floppies. As all platform variants of Globalview were based on the same Mesa code  (except device drivers), the existence of a 6085 version of GlobalView 3.x makes sense, as the hardware platform was still supported and in use in the 90-ies, even if no longer sold.

For accessing the bootfiles: i suppose that you mean by this that you want to manipulate the files at the file system layer. Well no, you can't see these files from VP. The reason is a bit complicated: the file systems of Pilot based systems are layered. The lower layer are Pilot files, which are simply collections of blocks and can be accessed only through the Pilot programming interface. Bootfiles are stored as Pilot files and there is a boot block page having pointers to these files. Each Pilot based operating system builds its own file system on top of Pilot files (MFile for Tajo aka XDE, NSFile for BWS aka ViewPoint/Globalview), implementing hierarchical directories, named files etc. And as the boot files are not owned by these upper level file systems, they are simply not present in MFile or NSFile file trees. Only specialized tools (installation programs like Othello) can and do access the boot files.

Greetings, Hans

**Comment by iblowmymind** [comment]
Hmm, I see. I really hope someone would find the XDE6, or the MDE 6.x and 5.x, that would be really cool! Also something weird I discovered (a bug?) while trying to use Draco in conjunction with Dodo:
I was transferring, like, 3000+ pages in total (in forms of small files) from the Draco machine to my file server in Dodo, so that later I could extract them from the machine to the real filesystem, but in the middle the packages just... stopped? I remember an error related to an "invalid token" in the Dodo log (but silly me forgot to save a log!), but there was no more transfer, no error opcode in Draco (but it was frozen), and no packets transferring except maybe 1 or 2 occasional ones. Is this a bug related to Draco or Dodo, or maybe both? I'll try to recreate the scenario and see if the error occurs again, and provide you a log output hopefully.

**Comment by devhawala** [comment]
Hello,

the "invalid token" error is high probably a misbehaviour of the filing session handling in Dodo: as required by the Filing protocol, sessions not used for a while time out and get invalid. This time period is 60 seconds, but resetting the timer occurs in Dodo when a filing operation is started: so when an operation takes longer, the session may time-out before the return, so the next operation runs into the invalid session error.

I already handled this (or thought to) for the (de)serialization operations used when storing/retrieving folders (i.e. directory trees, potentially many files). As you encountered the same problem when copying many single files, i added the prolongation of the session at the end of other potentially long running procedures, in the hope that it will help for your problem.

This version was uploaded a few minutes ago to the Dodo project.

And by the way: this Dodo version also contains the new XNS Mailing support for XDE, StarOS and ViewPoint. In fact this is the major part and reason for doing the release.

But be warned: there is no support for Mails with GlobalView, as GVWin (and very probably the other GlobalView versions for SunOS and OS/2) uses newer versions of the MailTransport and Inbasket Courier protocols, and GlobalView is not willing to fall back to the older protocol versions supported by Dodo.
And i don't not know is Dodo will ever support the new protocol versions, as these are completely undocumented (no PDF with at least verbalized descriptions or the like) and what i have seen so far of the GV mail Courier data streams is reluctant to analysis...

I hope the session problems are solved and have fun sending mails :-)

Greetings, Hans

**Comment by iblowmymind** [comment]
Oh that's really helpful, thank you! I hope that Xerox will release the source tree for Pilot, Star OS, XDE (and the devtools), and eventually ViewPoint (and maybe even GVWIN's both emulator and software-side?) in the near future like they did with the Alto, as they serve no real purpose in the modern office world anyways, because I don't think anyone would want to go and use VP in 2020 for serious business purposes (assuming they're even aware of something like that exists) At least they could release some documentation or missing pre-compiled versions such as XDE 6 and Star OS 1-4! (I'm not really sure what Star OS is, as I only saw it on BitSavers and nowhere else and I don't think there's any practical way to run it as it requires a factoring file to even boot, but I'm assuming it's the pre-ViewPoint 8010 software. The lack of documentation is really bugging me!)

**Comment by devhawala** [comment]
Hello,

most StarOS floppy sets at Bitsavers are in fact not very useful, due to missing a boot floppy or other problems...
Except for the "Star-4.3.zip" package, where the first floppy is bootable by Darkstar and it has 2 product factoring floppies. Despite the name, these floppies contain a StarOS 5.0 system. The trick is to look into one of the floppy images XCU-1.IMD or XCU-2.IMD which are the product factoring disks: the readable header tells that the network id and the password to use. That's how i created my StarOS system which i used to test Dodo with StarOS. Install from floppies WS1.IMD to WS-6.IMD (this makes a few reboots while installing) and then boot from XCU-1.IMD (only one of the XCU floppies is needed) and follow the instructions, entering the network-id and password found in the floppy header. And do not forget to set the network number in the Dodo configuration accordingly or the StarOS system may discard the license when booting...

Just another thing: i just uploaded a new version of Dwarf which supports a fullscreen mode for Duchess. Just add the command line parameter `-fullscreen` and there you go.

Enjoy
Hans

**Comment by californiaelectric** [comment]
Hans, as always your work in this domain is awe inspiring- so glad to see others taking advantage of it.

With regard to an “official” release of canonical binaries or a source tree sadly I would not hold your breath... The moment anyone shows interest in something, lawyers see dollar signs - it doesn’t matter how old or “obsolete” the software or hardware products seem to the rational mind.

For what it’s worth, the big difference from a business perspective between alto and star/vp/gv is that alto was a research product from PARC, never sold, while star and the derivatives were from business units of Xerox, sold as products with very strict licensing (the “product factoring” hans mentions) safeguards.

I’m just thankful that what has been preserved has been, and thanks to Hans, the we can appreciate it on modern hosts.

**Comment by iblowmymind** [comment]
Yeah, I agree californiaelectric. And I didn't notice the floppy header, thanks for pointing that out, I'll try it out now! And thank you for the fullscreen build as well!

**Comment by iblowmymind** [comment]
Hello, this has pretty much turned into a discussion board but, a tiny question regarding GVWIN (and I'm asking here since I couldn't find any resources online nor in the helpdocs and the only place I could ask is here):
So I managed to get WfW 3.11 working inside DOSBox (weird setup, I know, but I used something called Win31DOSBox to run it, which suprisingly integrated copy/paste, drag and drop and printing with Wfw3.11 which I don't know how) then I installed GVWIN 2.1 using the packdesc swap to make sure everything got installed (which worked), and since it was running in the official GVWIN emulator I could use the built-in PS printer, or so I thought. Trying to print anything using the PS Printer resulted in a error saying "Cannot create directory on the host PC" (or something similar, unsure). Testing it in a different configuration gave the same error, so is it about DOSBox? Also something else I've noticed is that GVWIN copies over TrueType and PostScript fonts for its built-in fonts (such as Classic, Modern, Equation etc.) so would it be hard to implement these fonts in Dodo's printing service? (since Dodo's Interpress to PostScript (to PDF) conversion doesn't actually result with the proper fonts, which results in poor-looking images.) Thanks!

**Comment by californiaelectric** [comment]
I’ve never been able to get any printing to work under dosbox. It’s also painfully slow with regard to windows 3.11 and gvwin.

You’ll have much better results if you use virtualbox, including GVWin printing, wither to a real postscript printer or to PDFs, etc.

Getting XNS networking working in virtualbox can be a bit tricky but it’s doable.

**Comment by iblowmymind** [comment]
Hmm. I'm starting to think it's because of DOSBox's weird folder-disk mix of a file system? And honestly, I've experienced similar speeds with DOSBox and VirtualBox running WfW 3.11. And I've had lots of trouble running WfW 3.11 on both VirtualBox and VMWare, screwing up the drivers multiple times resulting in broken systems. I really shouldn't forget to take backups. And yes, it would technically work if you set up a Bridged network in VirtualBox (although I'd prefer keeping it Host-only since that's where I'll be running Dodo), and tried connecting the server. I'll experiment more with it.

**Comment by iblowmymind** [comment]
Status update: Pretty much given up on trying to set up the appropriate drivers for 3.11. On VMWare I had issues patching the SVGA driver, on VirtualBox it just doesn't even boot most of the time. I tried writing a document using MS Write, and was able to print it both to a PDF on my desktop and even to a real printer without any issues. (Thanks to Win31DOSBox's great printing integration). The issue is either caused by DOSBox, or a misconfiguration in GVWin. On the help docs, PS Printing appearantly saves the PS files to C:\GVWINPS. Tried creating that directory by hand, still got the same error.

**Comment by californiaelectric** [comment]
A few notes-

I would not waste time trying to get XNS or printing to work with dosbox. Dosbox is awesome, for sure, but those two things just won’t work.

If using vbox use Windows 98. GV works just fine up to 98. Follow one of the set up guides you find using google to set up drivers, etc. turn off the audio drivers. I find sound blaster emulation really messes with the network emulation during booting. It still isn’t stable enough to boot fully every time, but just restart the vm enough times and it will.

If you are using wifi, you’ll really need to use host only networking. Use hans dodo gateway on the host only interface vbox creates to talk to your dodo instance.

After installing GVwin , find the XNS network drivers on the gvwin install media. Using the win98 networking ui, bind the XNS protocols to the network interface vbox provides.

Hope this helps.

**Comment by iblowmymind** [comment]
Thank you a lot! And yes, I did try it with versions later than W3.11 before, but I find Windows 95 almost impossible to install on modern VMs (without going in and patching it with a speedhack, even then it was unstable), and W98SE ran GVWIN pretty unstable as well, crashing often. On VMWare, Tools exist for W98 (afaik), you just need to install an additional PCI128 driver and sound works really well alongside the hardware. Creating a host-only network isn't a hassle either, but one thing I couldn't understand was figuring out the networking, since the networking thing on setup only provides two options: a Novell network and a Microsoft network, but I couldn't either to work on my testing. But I'll try out your advice! Thank you!

**Comment by devhawala** [comment]
Hello,

i had a quick look at the IPPROC.PS IP-to-PS conversion module of GVWin 2.1, which is also used by the Dodo print service. It seems that it defines the font mapping of Xerox fonts to PS fonts as hard coded translation. There is also some font-related mapping of the Xerox character sets. Not easy to understand at a glance.

It is however a fixed mapping. So odd-looking PS or PDF results may come from missing or not embedded fonts or from an "incomplete" font mapping. Maybe changing/rewriting the default mapping in IPPROC.PS can be adapted to use the set of pre-defined PS fonts to circumvent the font embedding problem. Fonts have always been non trivial things in my (commercial/professional) projects, source for more or less funny adventures (and/or loud cursing if you ask my colleages).

I think i will have a look at this, one of these days (or another day)...
Greetings, Hans

**Comment by californiaelectric** [comment]
Yes hans, when I looked into this myself I found the font substituting code within the .ps files in the GVWin printing section.

Rather than attempt to rewrite that, I think a more straightforward approach would be to chain to an additional post processor to substitute the fonts, and then perhaps to PDF to modernize the whole thing.

**Comment by iblowmymind** [comment]
Hello,
To confirm, yes, I have found a font conversion chart inside GVWIN's directory, that gave me the initial idea. The file should be called "IP2PS.FNT", and there are even fonts there that don't even exist and have been commented out. Interesting. And I do agree californiaelectric, that could be a better approach. There's also a "GDIPRNT.FNT", which's used for the GDI Print server, which contains entries for the Interpess name, TrueType name and the DOS filename. That could be helpful too.

**Comment by californiaelectric** [comment]
Practically speaking I have found it easiest on my XNS network - which includes two “real” 6085 machines - to copy files to a shared folder on the file service called “To be printed” and using the windows printer support from GVWin on Windows 98, printing all documents from there to my networked LaserJet. Yes the VP docs first get converted to GV but it’s very high fidelity. Once you get GVwin working well under Windows in an emulator you might try that approach as well.

**Comment by iblowmymind** [comment]
Hello. The thing is, I don't have any supported printers that work with Windows 98 in order to print them in the first place. I have a LaserJet at home, but it's just too new to support Windows 98 The oldest it goes is Server 2003. Besides, VMWare doesn't support virtual printers for Windows 98 in the first place, so that's a loss. Thank you for the suggestion though!

**Comment by iblowmymind** [comment]
Quick thing: I tried installing Smalltalk-80 DV6 on the emulated Draco XDE 5 installation. Sadly, I think all of the floppies are corrupt, as they can't be read by the emulator (except the microcode floppy, which hangs the system on read, so that's corrupt too-). I tried using IMDU to convert them to raw images then doing the extractXeroxFloppy (somewhere on the Bitsavers repo, I don't remember where) but they gave errors too, so I'm pretty sure it's corrupt.

**Comment by devhawala** [comment]
Hello,

yes, the floppy set for Smalltalk-80 at Bitsavers is broken as the first track is missing on all floppies, making it really difficult even to original Xerox OSes to read the floppies...

Even if you could read those floppies, you wouldn't be able to run Smalltalk-80 with Draco as the Dwarf-emulators can only run Mesa based things, notably only Pilot based OSes using the Mesa instructions of PrincOps 4.0 or the younger "MDS-releaved" successor. In that sense Draco emulates an 6085 machine with the Mesa microcode pre-loaded.

But the Smalltalk-80 DV6 environment can be installed only on a 6085 with 8K microcode store, as it uses a special microcode implementing both Mesa and Smalltalk instructions, where the Smalltalk microcode closely works with the Smalltalk support program (written in Mesa!) running on XDE. So one would need a real 6085 with added microcode store or a new/different 6085 emulator working at microcode level (as [Darkstar](https://github.com/livingcomputermuseum/Darkstar) does it for a Dandelion = 8010 machine) to have a chance to run Smalltalk-80 DV6 for 6085...

In short: Draco will not be able to run Smalltalk-80 environment.

BUT:
- if you "simply" want to run Smalltalk-80 DV6, it is already there and ready to run with my ST80 emulator project also at [github](https://github.com/devhawala/ST80), which provides a standalone Smalltalk-80 virtual machine as well as the DV6 image and environment created from the same Bitsavers floppy set (i remember to have patched Dwarf to get past the missing first track problem).

- so give [ST80](https://github.com/devhawala/ST80) a try if you want to play with some of the - vintage but original - Xerox Smalltalk-80 systems, preserved at the Bitsavers or archive.org sites.

Greetings, Hans

**Comment by iblowmymind** [comment]
Oh wait, I forgot that the machine wasn't microcoded. And I did try the ST80, I just wanted to inspect how it would run on a real 1186. Sorry about that! I was just diving in the bitsavers archive, looking at what I could find and throwing it at the machine to see what it actually is, as the lack of a proper software timeline (for example, was there any VP versions newer than 2.x or Star OS 5.2?) is pretty hard.

**Comment by kossow** [comment]
I checked with Josh, and there was a problem with the ST80 images, I redumped them and what is up there now works.
